src/overlays/actors/ovl_En_Bigpo/z_en_bigpo.c

```diff
@@ -94,161 +152,1305 @@ static DamageTable D_80B65044 = {
+}
+
+void EnBigpo_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnBigpo* this = (EnBigpo*)thisx;
+    s32 pad;
+    ColliderCylinder* thisCollider;
+
+    if ((this->actor.flags & 0x2000) == 0x2000) {
+        this->hoverHeightCycleTimer = 0;
+        this->savedHeight = this->actor.world.pos.y;
+    }
+
+    if (EnBigpo_ApplyDamage(this, globalCtx) == 0) {
+        if ((this->actor.isTargeted) && (this->actionFunc != EnBigpo_WarpingOut) &&
+            ((this->collider.base.acFlags & AC_HARD) == 0) && (this->actor.category == ACTORCAT_ENEMY)) {
+            this->unk20C += 1;
+        } else {
+            this->unk20C = 0;
+        }
+        if (this->unk20C == 0x28) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_PO_LAUGH);
+            EnBigpo_SetupWarpOut(this);
+        }
+    }
+
+    this->actionFunc(this, globalCtx);
+    if ((this->actionFunc != EnBigpo_SpawnPoCutscene6) && (this->actionFunc != EnBigpo_SpawnPoCutscene4)) {
+        Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+    }
+    if (this->actionFunc == EnBigpo_LanternFalling) {
+        Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 0.0f, 27.0f, 60.0f, 4);
+    }
+
+    if (this->actor.draw == EnBigpo_DrawScoopSoul) {
+        Actor_SetHeight(&this->actor, -10.0f);
+    } else {
+        Actor_SetHeight(&this->actor, 42.0f);
+    }
+
+    EnBigpo_UpdateColor(this);
+    EnBigpo_FlickerLanternLight(this);
+
+    this->actor.shape.shadowAlpha = this->mainColor.a;
+    thisCollider = &this->collider;
+    Collider_UpdateCylinder(&this->actor, thisCollider);
+    if ((this->collider.base.ocFlags1 & OC1_ON)) {
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &thisCollider->base);
+    }
+    if ((this->collider.base.atFlags & AT_ON)) {
+        CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &thisCollider->base);
+    }
+    if ((this->collider.base.acFlags & AC_ON)) {
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &thisCollider->base);
+    }
+
+    if (this->unk21C > 0.0f) {
+        Math_StepToF(&this->unk21C, 0.0f, 0.05f);
+        if (this->mainColor.a != 0xFF) { // NOT fully visible
+            if (this->mainColor.a * 0.003921569f < this->mainColor.a) {
+                this->unk21C = this->mainColor.a * 0.003921569f;
+            }
+        }
+        this->unk220 = ((this->unk21C + 1.0f) * 0.5f);
+        this->unk220 = (this->unk220 > 1.0f) ? 1.0f : this->unk220;
+    }
+}
+
+// alt update func: the revealed fires under dampe's house
+void EnBigpo_UpdateFire(Actor* thisx, GlobalContext* globalCtx) {
+    EnBigpo* this = (EnBigpo*)thisx;
+    // bilboarding the fire toward player camera?
+    this->actor.shape.rot.y = BINANG_ROT180(func_800DFCDC(ACTIVE_CAM));
+    this->actionFunc(this, globalCtx);
+}
+
+s32 EnBigpo_OverrideLimbDraw2(struct GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                              struct Actor* actor, Gfx** gfx) {
```

Comment:
You would probably need to run the formatter again.
```suggestion
s32 EnBigpo_OverrideLimbDraw2(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
                              Actor* actor, Gfx** gfx) {
```